docs/design/rosbag2_record_replay_service.md

```diff
@@ -0,0 +1,131 @@
+# Rosbag2 record and replay service
+
+## Context
+
+In [ros2/rosbag2#773](https://github.com/ros2/rosbag2/issues/773), it was recommended that record service requests feature should be added to rosbag2.  
+
+This design describe how to implement this feature based on current architecture and how to extend current command parameter.  
+
+## How to save record service message
+
+Service introspection ([ros2/ros2#1285](https://github.com/ros2/ros2/issues/1285)) had been implemented. All service messages (requests and responses) are published to service event topic which is hidden topic named as `_service_event` under service name (e.g. /add_two_ints/_service_event). So all messages in this topic can be recorded like general topic.
+
+## How to distinguish service event topic in all recorded topics
+
+Service event topic is saved like general topic. But they should use different process after reading from recording file. So they should be distinguished by topic name.
+
+- General topic  
+  
+  [NameSpace/]TopicName
+
+- Hidden topic  
+  
+  [NameSpace/]_TopicName
+
+- Service event topic
+  
+  [NameSpace/]ServiceName/_service_event
+
+The following criteria will be used to determine a service event topic:
+1. The topic ends with `/_service_event`
+2. The topic type name matches  `*/srv/*_Event` (e.g. `example_interfaces/srv/AddTwoInts_Event`)
+
+### Expand the 'record' command
+
+Add 5 parameters.
+
+- `--services ServiceName1 [ServiceName2 ...]`
+  
+    Specify which services will be recorded. User must specify at least one service.
+
+- `--all-topics`
+
+    Record all topics. Hidden topics including service event topics are excluded.
+
+- `--all-services`
+
+    Record all service event topics.
+
+- `--exclude-topics`
+
+    Rename from `--exclude`. Exclude topics containing provided regular expression.
+
+- `--exclude-services`
+
+    Exclude services containing provided regular expression.
+
+The description of the relevant parameter behavior.
+
+| Parameter | Description |
+| :-- | :--|
+|--include-hidden-topics| Record all hidden topics. Include service event topic. |
```

Comment:
this seems to be inconsistency for me.

as described above, service introspection topic is not one of the hidden topics, but this option includes the service event topic? as far as i see from the current discussion, this should not be included? just checking.

Reply:
I check discussion history. MichaelOrlov's comments (https://github.com/ros2/rosbag2/pull/1359#issuecomment-1574575492) decide this behavior.  You also agreed this at https://github.com/ros2/rosbag2/pull/1359#issuecomment-1575117306.  But emersonknapp gave different opinion two weeks ago (https://github.com/ros2/rosbag2/pull/1359#discussion_r1231528515).

Reply:
After giving it some thought, I've decided to further discuss this point at https://github.com/ros2/rosbag2/pull/1359#discussion_r1246072371